.github/workflows/test.yml

```
@@ -41,13 +51,14 @@ jobs:
 
   unit:
     name: Unit tests
-    runs-on: ubuntu-latest
+    runs-on: ${{ matrix.os }}
     strategy:
       fail-fast: false
       matrix:
         python-version:
         - "3.13"
         - "3.14"
+        os: [windows-latest, macos-latest, ubuntu-latest]
```

```suggestion
        python-version: ["3.13", "3.14"]
        os: [windows-latest, macos-latest, ubuntu-latest]
```
Is there a way to remove this duplication?

---

Dynamic matrix values could be used, which consists in a first job setting up matrices' values, making them as output values, and then the actual jobs use those outputs as matrix values. See e.g. [this stackoverflow answered question](https://stackoverflow.com/questions/65056670/is-it-possible-to-have-a-dynamic-strategy-matrix-in-a-workflow-in-github-actions).

---

I don't think the extra complexity is worth it.